src/stress/device/buffer_allocation.spec.ts

```diff
@@ -0,0 +1,27 @@
+export const description = `
+Stress tests for allocation of GPUBuffer objects through GPUDevice.
+`;
+
+import { makeTestGroup } from '../../common/framework/test_group.js';
+import { GPUTest } from '../../webgpu/gpu_test.js';
+
+export const g = makeTestGroup(GPUTest);
+
+g.test('concurrent_buffer_allocation')
+  .desc(`Tests allocation of many concurrent GPUBuffer objects.`)
+  .unimplemented();
+
+g.test('continuous_buffer_allocation_with_destroy')
+  .desc(
+    `Tests allocation and destruction of many GPUBuffer objects over time. Objects
+are sequentially created and destroyed over a very large number of iterations.`
+  )
+  .unimplemented();
+
+g.test('continuous_buffer_allocation_with_gc')
```

Comment:
FWIW: Our garbage collectors don't really understand the memory cost of these objects (since they're in GPU memory, not CPU memory), so these tests might not end up behaving the way you might expect. However you can use the `attemptGarbageCollection` helper to try to incur GC.

Reply:
Actually incurring GC would not be a hard expectation of these tests, the only expectation would be that the test doesn't blow up due to lack of explicit disposal. My intent was to reflect as accurately as possible the UA's behavior when confronted with an application that indiscriminately creates and drops lots of objects.